code/controllers/CMSMain.php

```diff
@@ -88,8 +97,24 @@ public function init() {
 
 		CMSBatchActionHandler::register('publish', 'CMSBatchAction_Publish');
 		CMSBatchActionHandler::register('unpublish', 'CMSBatchAction_Unpublish');
-		CMSBatchActionHandler::register('delete', 'CMSBatchAction_Delete');
-		CMSBatchActionHandler::register('deletefromlive', 'CMSBatchAction_DeleteFromLive');
+		
+
+		// Check legacy actions
+		$legacy = $this->config()->enabled_legacy_actions;
```

Comment:
There's a config parameter which already exists on CMSBatchActionHandler. 

eg.
$registered = Config::inst()->get('CMSBatchActionHandler', 'batch_actions');
if(!array_key_exist('delete', $registered)) { // enable archive }

Then you can add the deprecation warning to the CMSBatchAction_Delete(FromLive) classes instead 

In 4.0 they can all just be added to the static config and live happily ever after \o/